xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/RecentlyUpdated.xml

```diff
@@ -0,0 +1,1744 @@
+        <unmodifiable>0</unmodifiable>
+        <values>long|short|default|forbid</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </cache>
+      <code>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>const MACRO_SERVICE_URL = "$xwiki.getURL('Confluence.Macros.RecentlyUpdatedService', 'get')";
+#[[
+
+require(['jquery'], $ =&gt; {
+  $(() =&gt; {
+    const $macros = $('.recently-updated-macro');
+
+    /*
+     * Query new results and append them to the page
+     */
+    const fetchMoreResultsHTML = async ($macro) =&gt; {
+      const urlParams = new URLSearchParams({
+        'xpage': 'plain',
+        'outputSyntax': 'plain',
+        'macroOptions': $macro.data('options')
+      });
+      const url = `${MACRO_SERVICE_URL}?${urlParams.toString()}`;
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error('An error occured fetching new results.');
+      }
+      // Append html results directly to the page
+      const html = await response.text();
+      const template = document.createElement('template');
+      template.innerHTML = html;
+      $macro.find('ul.results')[0].appendChild(template.content);
```

Comment:
Because it's much much faster to parse html using native js APIs (and parsing it out of the document) than it is to be parsed with jquery!
But if we need to support for IE11 I'm going to change html templates for document fragments, which allow to do a similar job and is supported by IE.

Reply:
Finally directly append html inside parent element, as document fragments cannot directly parse content from stringified html.